src/main/java/randoop/reflection/RawSignature.java

```diff
@@ -44,7 +44,9 @@ public RawSignature(
       String classname,
       String name,
       Class<?>[] parameterTypes) {
-    assert !Objects.equals(packageName, "");
+    // In Java 8 the unnamed package name is null.
+    // In Java 9 the unnamed package name is "".
+    // assert !Objects.equals(packageName, "");
```

Comment:
I would delete this rather than commenting it out, and move the documentation to the method header.
You might also leave it in, but java-version-specific.